### ISSUE TITLE
Angular 5 compile errors

### ISSUE THREAD

**Issue by ChrisMBarr** [stevepapa/ng-autosize#15]
When using Angular 5, Typescript 2.6.x, and enabling the `noUnusedLocals` compiler option in `tsconfig.json`, it produces this error when attempting to build:

```  
ERROR in
node_modules/ng-autosize/src/autosize.directive.ts(12,13): error TS6133: '_lastHeight' is declared but its value is never read.
node_modules/ng-autosize/src/autosize.directive.ts(34,14): error TS6133: 'textArea' is declared but its value is never read.
node_modules/ng-autosize/src/autosize.directive.ts(42,11): error TS6133: 'textArea' is declared but its value is never read.
node_modules/ng-autosize/src/autosize.directive.ts(53,52): error TS2345: Argument of type 'null' is not assignable to parameter of type 'string | undefined'.
```

This seems related to #12 - if that were fixed, this problem would likely just go away.  However, those error make some good suggestions for code cleanup.

**Comment by shyamal890** [comment]
+1

**Comment by smasala** [comment]
+1

**Comment by bardaxx** [comment]
Solved in #17

**Comment by danielszuk** [comment]
+1
